omnibus/files/private-chef-upgrades/001/037_elasticsearch_to_opensearch_migration.rb

```diff
@@ -0,0 +1,36 @@
+define_upgrade do
+    if Partybus.config.bootstrap_server
+  
+      es = Partybus.config.running_server["private_chef"]["opensearch"]
+      version = File.read("/opt/opscode/embedded/opensearch/VERSION")
+      # Run these migrations only if elasticsearch has been enabled in the config
+      # We do not want to run this if solr is enabled
+  
+      if es["enable"] && Gem::Version.new(version) == Gem::Version.new("1.2.4")
+  
+        must_be_data_master
+
+        sleep 30
+  
+        log "Migrating indexed search data..."
+        
+        start_services(["opensearch",
+          "nginx",
+          "oc_bifrost",
+          "opscode-erchef",
+          "postgresql",
+          "redis_lb"])
+        run_command("chef-server-ctl reindex -a -d -t")
+        # Not including postgres in services here
+        # since external postgres cannot be managed
+        # via chef-server-ctl
```

Comment:
will `stop_services` fail in case of external? 
We do start it above, and I'm thinking if that were a problem it would have failed there.

Reply:
No, Marc. It is not creating any problem.